packages/orderbook/src/order_set.ts

```diff
@@ -0,0 +1,66 @@
+import { APIOrder } from '@0x/connect';
+
+import { utils } from './utils';
+
+export class OrderSet {
+    private readonly _map: Map<string, APIOrder>;
+    constructor(orders: APIOrder[] = []) {
+        this._map = new Map();
+        (this as any)[Symbol.iterator] = this.values;
+        for (const order of orders) {
+            this.add(order);
+        }
+    }
+
+    public size(): number {
+        return this._map.size;
+    }
+
+    public add(item: APIOrder): void {
+        const orderHash = utils.getOrderHash(item);
+        (item.metaData as any).orderHash = orderHash;
```

Comment:
Why do we need this `any` cast?

Reply:
We do not define what is inside of the metaData as per the SRA spec. Launch kit returns `orderHash` but it is not enforced by the spec and Radar does not return this.